services/src/handlers/workflows.rs

```diff
@@ -240,19 +257,299 @@ async fn get_workflow_provenance<C: Context>(
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::I32(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::F32(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        geoengine_operators::engine::TypedRasterQueryProcessor::F64(p) => {
+            raster_stream_to_geotiff(
+                file_path,
+                p,
+                query_rect,
+                query_ctx,
+                no_data_value,
+                request_spatial_ref,
+                tile_limit,
+            )
+            .await
+        }
+        _ => return Err(error::Error::RasterDataTypeNotSupportByGdal),
+    }
+    .map_err(error::Error::from)?;
+    Ok(())
+}
+
+async fn create_dataset<C: Context>(
+    info: RasterDatasetFromWorkflow,
+    file_path: std::path::PathBuf,
+    result_descriptor: &geoengine_operators::engine::RasterResultDescriptor,
+    ctx: C,
+    session: <C as Context>::Session,
+) -> Result<geoengine_datatypes::dataset::DatasetId> {
+    let dataset_id = InternalDatasetId::new().into();
+    let dataset_definition = DatasetDefinition {
+        properties: AddDataset {
+            id: Some(dataset_id),
+            name: info.name,
+            description: info.description.unwrap_or_default(),
+            source_operator: "GdalSource".to_owned(),
+            symbology: None,  // TODO
+            provenance: None, // TODO
+        },
+        meta_data: MetaDataDefinition::GdalStatic(GdalMetaDataStatic {
+            time: Some(info.query.time_interval),
+            params: GdalDatasetParameters {
+                file_path,
+                rasterband_channel: 1,
+                geo_transform: GeoTransform::new(
+                    info.query.spatial_bounds.upper_left(),
+                    info.query.spatial_resolution.x,
+                    -info.query.spatial_resolution.y,
+                ),
+                width: (info.query.spatial_bounds.size_x() / info.query.spatial_resolution.x).ceil()
+                    as usize,
+                height: (info.query.spatial_bounds.size_y() / info.query.spatial_resolution.y)
+                    .ceil() as usize,
+                file_not_found_handling: FileNotFoundHandling::Error,
+                no_data_value: result_descriptor.no_data_value,
+                properties_mapping: None, // TODO
+                gdal_open_options: None,  // TODO
```

Comment:
Please state what is `todo`

Reply:
09ce2d78b2cf0251d2c21f7396880121fb0286c6